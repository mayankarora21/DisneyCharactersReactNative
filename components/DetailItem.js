import { Text, StyleSheet } from "react-native";

const DetailItem = ({ character, keyObject }) => {
  return (
    <Text style = {styles.text}>
        {keyObject.visibleKey}: {character[keyObject.actualKey].length !== 0 ? character[keyObject.actualKey].join(", ") : `No ${keyObject.visibleKey}`}
    </Text>
  );
};

export default DetailItem;

const styles = StyleSheet.create({
    text: {
        borderBottomWidth: 1,
        padding: 10
    }
})