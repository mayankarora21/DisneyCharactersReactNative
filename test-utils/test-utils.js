import React from 'react'
import {render} from '@testing-library/react-native';
import store from '../store/store';
import { Provider } from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';

const AllTheProviders = ({children}) => {
  return (
    <Provider store = {store}>
      <NavigationContainer>
            {children}
      </NavigationContainer>
    </Provider>
  )
}

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}