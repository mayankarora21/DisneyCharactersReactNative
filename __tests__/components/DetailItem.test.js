import DetailItem from '../../components/DetailItem';
import {render, screen} from '../../test-utils/test-utils';

test('DetailItem should render correctly when films array is empty', () => {
    render(<DetailItem character={{name: 'CharacterName', films: []}} keyObject = {{actualKey: 'films', visibleKey: 'Films'}}/>)
    expect(screen.toJSON()).toMatchSnapshot();
})

test('DetailItem should render correctly when films array is not empty', () => {
    render(<DetailItem character={{name: 'CharacterName', films: ["Film1", "Film2"]}} keyObject = {{actualKey: 'films', visibleKey: 'Films'}}/>)
    expect(screen.toJSON()).toMatchSnapshot();
})