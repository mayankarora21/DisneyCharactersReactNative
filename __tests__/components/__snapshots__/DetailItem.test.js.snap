// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailItem should render correctly when films array is empty 1`] = `
<Text
  style={
    {
      "borderBottomWidth": 1,
      "padding": 10,
    }
  }
>
  Films
  : 
  No Films
</Text>
`;

exports[`DetailItem should render correctly when films array is not empty 1`] = `
<Text
  style={
    {
      "borderBottomWidth": 1,
      "padding": 10,
    }
  }
>
  Films
  : 
  Film1, Film2
</Text>
`;
